### PR TITLE
fix(component): make passport packages optional

### PR DESCRIPTION
**Pull request by RaghavaroraSF**
## Description

All passport packages are made optional, and the user should install them only if required.

BREAKING CHANGE:
change import path for specific passport packages

Fixes # GH-125

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [ ] Performed a self-review of my own code
- [x] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by yeshamavani — CHANGES_REQUESTED** [review]

File: README.md [hunk 1]
```diff
@@ -71,6 +71,7 @@ Once this is done, you are ready to configure any of the available strategy in t
 
 ### Oauth2-client-password
 
+In order to use it, run `npm install passport-oauth2-client-password`.
```
Comment (yeshamavani) [inline README.md:74]: Also mention about the specific provider binding required

**Review by yeshamavani — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -1008,6 +1012,7 @@ For accessing the authenticated AuthUser model reference, you can inject the CUR
 
 ### Google Oauth 2
 
+In order to use it, run `npm install passport-google-oauth20` or `npm install @types/passport-google-oauth20`.
```
[the remark this answers is not in the document]
Comment (RaghavaroraSF) [reply README.md:1015]: resolved

File: README.md (hunk 1 above)
Comment (RaghavaroraSF) [reply README.md]: resolved

**Review by yeshamavani — COMMENTED** [review]
The providers are binded which use the packages so this will not actually remove the dependency need

**Pull request closed without merging**